app.rb

```diff
@@ -0,0 +1,161 @@
+    print('Age: ')
+    age = gets.chomp
+    print('Name: ')
+    name = gets.chomp
+    print('Specialization: ')
+    specialization = gets.chomp
+    teacher = Teacher.new(name, age, specialization)
+    @persons.push(teacher)
+    puts "Teacher created successfully\n"
+  end
+
+  def add_book
+    puts '+++++++++ Add Book +++++++++'
+    print('Title: ')
+    title = gets.chomp
+    print('Author: ')
+    author = gets.chomp
+    book = Book.new(title, author)
+    @books.push(book)
+    puts "Book created successfully\n"
+  end
+
+  def create_rental
+    puts 'Select a book from the following list by number '
+    puts "\t\t+    No Book found, please Add a book  +" if @books.size.zero?
+    @books.each_with_index do |book, index|
+      puts "#{index}) Title: \"#{book.title}\", Author: #{book.author}"
+    end
+    book_index = gets.chomp.to_i
+
+    puts 'Select a person from the following list by number (not id)'
+    puts "\t\t+    No Person found, please Add a book  +" if @persons.size.zero?
+    @persons.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    person_index = gets.chomp.to_i
+
+    puts 'Date [YYYY-MM-DD] :'
+    date = gets.chomp
+    rental = Rental.new(date, @books[book_index], @persons[person_index])
+    @rentals.push(rental)
+
+    puts "Rental created successfully\n"
+  end
+
+  def list_rental
+    print 'ID of person: '
+    @persons.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+    id = gets.chomp.to_i
+    person = @persons.find { |p| p.id == id }
+    puts 'Rentals: '
+    person.rentals.each { |rental| puts "Date: #{rental.date}, Book: #{rental.book.title} by #{rental.book.author}" }
+  end
+
+  def create_person()
+    loop do
+      print 'Do you want to create a student (1) or a teacher (2)? [Input the number] :'
+      choose = gets.chomp
+      case choose.to_i
+      when 1
+        create_student
+        break
+      when 2
+        create_teacher
+        break
+      end
+      break unless choose.to_i == 1 || choose.to_i == 2
+    end
+  end
+
+  def message
+    p '======  Welcome to Our School Library Apps  ======'
+    p '=================================================='
+    puts 'Please choose an option by entering a number:'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/MethodLength
+  def run()
+    @rep = 0
+    loop do
+      message
+      @rep = gets.chomp
+      case @rep.to_i
+      when 1
+        list_books(@books)
+      when 2
+        list_person(@persons)
+      when 3
+        create_person
+      when 4
+        add_book
```

Comment:
* Kindly Make sure that the role of main.rb is only to start your code and ask the user for the initial choice of options (or to quit), so I strongly suggest you extract these pieces of code into your main.rb file as you are to [Pay special attention to the "Single-responsibility Principle" ](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/school_library_refactor.md#:~:text=Pay%20special%20attention%20to%20the%20%22Single%2Dresponsiblity%20Principle%22)